### DISCUSSION TITLE
Compact non-vertical tabs

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#846] [Ideas]
### What feature would you like to see?

I would like to see an option to have the tabs on the top bar but, inspired by the style Safari's compact mode provides. I am not particularily a fan of vertical tabs, now am I a fan of any kind of side bar which, vertical tabs needs.

### What platform are you seeing the problem on?

Windows

### Additional context

It would also be great if we had options to individually or/and globally set it so. The selected tab showing the page title and when you click on that tab, the search bar and current URL shows up within that tab.
 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/409_

**Comment by ghost** [comment]
Mind the typos, I created this past midnight.

But, on the point, I have always been a fan of how Safari does compact tabs and think the concept would do brilliantly with Zen's UI/UX design. 

Things to consider which I can think of off the top of my head:
- Have the search bar split into 'pills', either taking up all available width or having a set width and they all center.
- Have options to set all or individual 'pills' to show just the website favicon or favicon + title to save space.
- When double clicking on the tab's 'pill', it will turn into a search bar.
- When opening a new tab, add an X button at the start to close the tab when the search bar is visible and have that X button visible somewhere when the 'pill' is showing the website title.
- An option to hide the sidebar while in this compact mode.

**Comment by github-account1111** [comment]
This would make me ditch Edge in an instant.

Vertical tabs, while an improvement from traditional tabs, just feel clunky and out of place as a sidebar. They're either too wide and encroach on usable screen estate, or too narrow, and I don't even know which tab is which because all the text is cut off.

Safari's compact tabs solve this issue along with multiple others. They free up vertical space, just like vertical tabs do, but they are wide enough for the user to see what the tab says.

Compact tabs also feel much more intentional. At a glance, the user sees the names of all tabs with a bit more space for the current tab. Clicking on the current tab turns it into an address bar. It's a much smarter use of space. You don't need a sidebar at all, and you only need a *single* top bar (right now Zen has to have both).

Compact tabs are one of the *only two* things I liked about MacOS and miss a lot about it after going back to Windows. It's sad how much flack Apple got when they introduced it. It would've received so much more adoption if people weren't so averse to change, even when it's objectively good.
